packages/headless/src/test/mock-engine.ts

```diff
@@ -60,6 +60,10 @@ export interface MockSearchEngine
   extends SearchEngine<SearchAppState>,
     MockEngine {}
 
+/**
+ * Returns non-functionnal `SearchEngine`.
+ * To be used only for unit testing controllers, not for functionnal tests.
+ */
```

Comment:
Could also add "deprecate" if we want to remove them in the future

Reply:
IMO, if we deprecate them I don't think we should also document them. So far I prefer the idea of deprecating all of them.

Reply:
I would add a disclaimer: "For internal use only". 

Like making it obvious that we don't encourage people to use this...